### PR TITLE
Files changed

### PR DESCRIPTION
**Pull request by jrieken**
This PR covers the ground work for #106 and I would like get it out before it gets too big. Until now, it contains these changes

* replace the (flaky) file event watcher with a ```filesChanged``` route which drives the ```IFileWatcher```
* Add an IEventEmitter interface which allows OmniSharp to talk back to an editor (such that it can communicate the effects of changed files)
* Listen for some workspace events (project modifications) and send those events (in the shape of ```ProjectInformationResponse```-objects) to the connected editor
* Make ```ProjectInformationResponse``` also include the source files of a project such that an editor can use this information, for instance requesting diagnostics for more than the open files.

What still needs to be done is refining the file events (adding types ```added```, ```changed```, and ```deleted```) and make the ASP project system smarter such that it picks up new files.

### CODE REVIEW

**Comment by nosami** [comment]
I think right now, I'd be really happy if we could just add a new cs file to the workspace :) Could you add that in and we can add the other stuff later?

Some guy on jabbr was just complaining that he has to restart sublime to add a new file :P

File: src/OmniSharp/Roslyn/ProjectEventForwarder.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                        Task.Factory.StartNew(async () =>
+                        {
+                            await Task.Delay(500);
+
+                            lock (_lock)
+                            {
+                                _queue.Remove(fileNameAndEventKind);
+
+                                object payload = null;
+                                if (fileNameAndEventKind.Item2 != WorkspaceChangeKind.ProjectRemoved)
```
Comment (nosami) [inline src/OmniSharp/Roslyn/ProjectEventForwarder.cs]: Tuples suck in C#. What is Item2? Could we use an anonymous object instead?

Comment (davidfowl) [inline src/OmniSharp/Roslyn/ProjectEventForwarder.cs]: Yah or make a type.

File: src/OmniSharp/Roslyn/ProjectEventForwarder.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            _aspnet5Context = aspnet5Context;
+            _msbuildContext = msbuildContext;
+            _workspace = workspace;
+            _emitter = emitter;
+            _workspace.WorkspaceChanged += OnWorkspaceChanged;
+        }
+
+        private void OnWorkspaceChanged(object source, WorkspaceChangeEventArgs args)
+        {
+
```
Comment (davidfowl) [inline src/OmniSharp/Roslyn/ProjectEventForwarder.cs]: Extra space.

**Comment by jrieken** [comment]
@davidfowl - squashed and without extra space

**Pull request merged**